### DISCUSSION TITLE
Menu stacking issue

### DISCUSSION THREAD

**Discussion by spidertwin2** [chakra-ui/chakra-ui#2677] [Q&A] [answered]
Hi,

I am unable to position elements correctly.  The ViewIcon is visible on top of the dropdown menu, how should I send it back?

![img1](https://user-images.githubusercontent.com/29196888/100735440-3986b400-33f7-11eb-9eee-112033cfdf06.png)
![img2](https://user-images.githubusercontent.com/29196888/100735443-3ab7e100-33f7-11eb-8b57-579acc9c0adb.png)

signup.js
```
<InputGroup size="md">
  <Input
    {...field}
    id="password"
    type={show ? "text" : "password"}
    placeholder="Enter password"
    isRequired={true}
  />
  <InputRightElement h="1rem">
    <IconButton
      mr="0"
      variant="ghost"
      colorScheme="gray.800"
      aria-label="Show Hide Password"
      icon={show ? <ViewOffIcon /> : <ViewIcon />}
      onClick={handleShow}
    />
  </InputRightElement>
</InputGroup>
```
navbar.js
```
<Menu m="auto" z-index="1">
  <MenuButton as={Button} rightIcon={<ChevronDownIcon />}>
    MyAccount
  </MenuButton>
  <MenuList>
    <MenuItem>Profile</MenuItem>
    <MenuItem>Community</MenuItem>
    <MenuItem>Resources</MenuItem>
    <MenuItem>Discussion</MenuItem>
    <MenuItem>Stats</MenuItem>
    <MenuDivider />
    <MenuItem>Settings</MenuItem>
    <MenuItem>SignOut</MenuItem>
  </MenuList>
</Menu>
```
Thank you :)

**Comment by chr-ge** [comment, marked as answer]
Try adding `zIndex="dropdown"` to the `Menu` instead of `z-index="1"`.

**Reply by spidertwin2** [reply]
I should have shared the whole code earlier, for a better perspective.  Adding `position="relative"` and `zIndex="dropdown"` to the outermost parent <Flex> solved it! Thanks :pray: 

```
<Flex
      direction="row"
      w="100%"
      p="3% 10%"
      justify="space-between"
      color={useColorModeValue("gray.800", "white")}
      position="relative"
      zIndex="dropdown"
    >
      <Flex>
        <BrandName>
          <Link href="/">
            <a>App</a>
          </Link>
        </BrandName>
      </Flex>
      <Flex w="170px" justify="space-between">
        <DarkModeSwitch />
        <Menu m="auto">
          <MenuButton as={Button} rightIcon={<ChevronDownIcon />}>
            MyAccount
          </MenuButton>
          <MenuList>
            <MenuItem>Profile</MenuItem>
            <MenuItem>Community</MenuItem>
            <MenuItem>Resources</MenuItem>
            <MenuItem>Discussion</MenuItem>
            <MenuItem>Stats</MenuItem>
            <MenuDivider />
            <MenuItem>Settings</MenuItem>
            <MenuItem>SignOut</MenuItem>
          </MenuList>
        </Menu>
      </Flex>
    </Flex>
```

**Reply by derrick-null** [reply]
You saved my day. This issue bothered me for 2 hours. My scenario is I have customized dropdown composites with an Input and a List. Below the dropdown is a table with a checkbox that can select all items of the table. The checkbox is always displayed. Following your approach, I finally solved this issue.

**Reply by rishabhjaiz** [reply]
Curious how does `zIndex="dropdown"` solve this, also the type sense does show dropdown as a valid option how did you came to know about this?
<img width="529" alt="Screenshot 2024-06-17 at 2 32 15 PM" src="https://github.com/chakra-ui/chakra-ui/assets/51243003/69dcf0bb-dc69-4661-b762-88451c48c085">
